src/parser/node.js

```diff
@@ -34,43 +34,40 @@ class Node {
   }
 }
 
-pp.startNode = function () {
-  return new Node(this, this.state.start, this.state.startLoc);
-};
-
-pp.startNodeAt = function (pos, loc) {
-  return new Node(this, pos, loc);
-};
-
-function finishNodeAt(node, type, pos, loc) {
-  node.type = type;
-  node.end = pos;
-  node.loc.end = loc;
-  if (this.options.ranges) node.range[1] = pos;
-  this.processComment(node);
-  return node;
-}
+export class NodeUtils extends UtilParser {
+  startNode() {
+    return new Node(this, this.state.start, this.state.startLoc);
+  }
 
-// Finish an AST node, adding `type` and `end` properties.
+  startNodeAt(pos, loc) {
+    return new Node(this, pos, loc);
+  }
 
-pp.finishNode = function (node, type) {
-  return finishNodeAt.call(this, node, type, this.state.lastTokEnd, this.state.lastTokEndLoc);
-};
+  // Finish an AST node, adding `type` and `end` properties.
 
-// Finish node at given position
+  finishNode(node, type) {
+    return this.finishNodeAt(node, type, this.state.lastTokEnd, this.state.lastTokEndLoc);
+  }
 
-pp.finishNodeAt = function (node, type, pos, loc) {
-  return finishNodeAt.call(this, node, type, pos, loc);
-};
+  // Finish node at given position
 
+  finishNodeAt(node, type, pos, loc) {
```

Comment:
Not sure why this was initially broken into both `pp.finishNodeAt` and `function finishNodeAt`. Seems to work fine this way.